syslog_rfc5424_formatter.py

```diff
@@ -36,24 +36,17 @@ class RFC5424Formatter(logging.Formatter, object):
     the format string that you pass in the constructor is only
     applied to the message body (and should typically just be %(message)).
 
-    The '- -' sections in the resulting message are the "msg ID" and
-    "Structured-Data" Elements, respectively
-
-    MSGID (Description from RFC5424):
-       The MSGID SHOULD identify the type of message.  For example, a
-   firewall might use the MSGID "TCPIN" for incoming TCP traffic and the
-   MSGID "TCPOUT" for outgoing TCP traffic.  Messages with the same
-   MSGID should reflect events of the same semantics.  The MSGID itself
-   is a string without further semantics.  It is intended for filtering
-   messages on a relay or collector.
-   The NILVALUE SHOULD be used when the syslog application does not, or
-   cannot, provide any value.
-
    Stuctured Data Example:
         [exampleSDID@32473 iut="3" eventSource="Application" eventID="1011"]
     '''
-    def __init__(self, *args, **kwargs):
+    def __init__(self, appname='python', procid='-', msgid='-', data='-', *args, **kwargs):
         self._tz_fix = re.compile(r'([+-]\d{2})(\d{2})$')
+        self.__defaults ={
+            'appname': appname,
+            'procid': procid,
+            'msgid': msgid,
+            'data': data
```

Comment:
not clear what you're trying to achieve with `data`

the data is always just the message body...

Reply:
data is the STRUCTURED-DATA (Currently always "-") part which while I agree I couldn't see a use for a default, I also couldn't see a reason not to add it encase someone did want it for some reason.
Happy not to have a default for data if you prefer?

Reply:
Ah; I would probably name it `structured_data`. If we're going to support it being something other than empty, we would probably want to take a dict or something and encode it since the encoding is a little obnoxious and syslog servers (at least syslog-ng) get quite angry when you send malformed SD sections.

Reply:
Happy with rename. I was going to look at allowing it to be a dict later, but can look at it now. I do agree that is has to be right else syslog servers will reject it.